addons/mail/models/mail_thread.py

```diff
@@ -980,16 +986,15 @@ def _detect_loop_headers(self, msg_dict):
     @api.model
     def _detect_write_to_catchall(self, msg_dict):
         """Return True if directly contacts catchall."""
-        catchall_alias = self.env['ir.config_parameter'].sudo().get_param("mail.catchall.alias")
-        email_to = msg_dict['to']
-        email_to_localparts = [
-            e.split('@', 1)[0].lower()
-            for e in (tools.email_split(email_to) or [''])
+        catchall_aliases = self.env['mail.alias.domain'].search([]).mapped('catchall_email')
```

Comment:
shouldn't we call email_normalize on `catchall_email` ? 

(the locat parts are cleaned with `_sanitize_alias_name`, but not the domain part)